mods/bucket/init.lua

```diff
@@ -153,6 +153,22 @@ minetest.register_craftitem("bucket:bucket_empty", {
 			minetest.add_node(pointed_thing.under, {name="air"})
 
 			return ItemStack(giving_back)
+		else
+			local node_def = minetest.registered_nodes[node.name]
+			if node_def then
+				-- Buckets will run a node's on_punch function if it is not liquid.
+				if node_def.on_punch then
+					giving_back = node_def.on_punch(
```

Comment:
`on_punch` doesn't define return values in the api.

Btw, you could merge the if's `if node_def and node_def.on_punch`, but that's not a big deal.